app/src/main/java/com/mapbox/maps/testapp/examples/snapshotter/DataDrivenMapSnapshotterActivity.kt

```diff
@@ -38,48 +37,50 @@ class DataDrivenMapSnapshotterActivity : AppCompatActivity() {
       .resourceOptions(MapboxOptions.getDefaultResourceOptions(this))
       .build()
 
-    snapshotter = Snapshotter(this, snapshotMapOptions)
-    snapshotter.setCameraOptions(
-      CameraOptions.Builder()
-        .center(Point.fromLngLat(-94.0, 15.0))
-        .zoom(5.0)
-        .padding(EdgeInsets(1.0, 1.0, 1.0, 1.0))
-        .build()
-    )
-    snapshotter.setUri(Style.OUTDOORS)
-    snapshotter.start(object : Snapshotter.SnapshotReadyCallback {
-      override fun onStyleLoaded(style: Style) {
-        showToast("Map load style success!")
-
-        // GeoJSON earthquake source using low level style API
-        val properties = HashMap<String, Value>()
-        properties["type"] = Value.valueOf("geojson")
-        properties["data"] = Value.valueOf(EARTHQUAKE_SOURCE_URL)
-        val sourceResult = style.addStyleSource(EARTHQUAKE_SOURCE_ID, Value.valueOf(properties))
-        if (sourceResult.isError) {
-          throw RuntimeException(sourceResult.error)
-        }
-
-        // Heatmap layer using high level style PI
-        style.addLayer(getHeatmapLayer())
-      }
+    snapshotter = Snapshotter(this, snapshotMapOptions).apply {
+      setStyleListener(object : SnapshotStyleListener {
+        override fun onDidFinishLoadingStyle(style: Style) {
+          Toast.makeText(
+            this@DataDrivenMapSnapshotterActivity,
+            "Map load style success!",
+            Toast.LENGTH_LONG
+          ).show()
 
-      override fun onSnapshotCreated(snapshot: Expected<MapSnapshotInterface?, String?>) {
-        if (snapshot.isError) {
-          showToast("ERROR! ${snapshot.error}")
-        } else {
-          val image = snapshot.value!!.image()
-          val configBmp: Bitmap.Config = Bitmap.Config.ARGB_8888
-          val bitmap: Bitmap = Bitmap.createBitmap(image.width, image.height, configBmp)
-          val buffer: ByteBuffer = ByteBuffer.wrap(image.data)
-          bitmap.copyPixelsFromBuffer(buffer)
+          // GeoJSON earthquake source using low level style API
+          val properties = HashMap<String, Value>()
+          properties["type"] = Value.valueOf("geojson")
+          properties["data"] = Value.valueOf(EARTHQUAKE_SOURCE_URL)
+          val sourceResult = style.addStyleSource(EARTHQUAKE_SOURCE_ID, Value.valueOf(properties))
+          if (sourceResult.isError) {
+            throw RuntimeException(sourceResult.error)
+          }
```

Comment:
Is it intentional to use low level style API?

Reply:
I did not touch that logic, I guess author of that example did it intentionally.